src/arrayMethodSort.js

```diff
@@ -4,8 +4,38 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  const compareAsStrings = (a, b) => {
+    const stringA = String(a);
+    const stringB = String(b);
+
+    if (stringA > stringB) {
+      return 1;
+    } else if (stringA === stringB) {
+      return 0;
+    } else {
+      return -1;
+    }
+  };
+
+  [].__proto__.sort2 = function(compareFunction = compareAsStrings) {
+    let counter;
+
+    do {
+      counter = 0;
+
+      for (let i = 1; i < this.length; i++) {
+        const previous = this[i - 1];
+        const current = this[i];
+
+        if (compareFunction(previous, current) > 0) {
+          counter++;
+          this[i - 1] = current;
+          this[i] = previous;
+        }
+      }
+    } while (counter > 0);
```

Comment:
think about a more optimized solution

Reply:
here is useful information on this topic
https://www.geeksforgeeks.org/bubble-sort/

Reply:
Please, review @silchyn comment, instead of counting changes and going through the whole array each time you can use boolean variable and stop the algorithm if the inner loop didn’t cause any swap.